src/SfxWeb/src/app/Models/eventstore/timelineGenerators.ts

```diff
@@ -186,6 +188,45 @@ export abstract class TimeLineGeneratorBase<T> {
          throw new Error('NotImplementedError');
     }
 
+    getSimultaneousEventsForEvent(configs: IConcurrentEventsConfig[], events: DataSet<DataItem>) : IConcurrentEvents[] {
+        /*
```

Comment:
so this should return data a bit differently. we should return only back the events we care about and not all the matching events.

I.E
The objects returned should be for saying a config of ApplicationProcessExit, we should see a list of ApplicationProcessExist events with concurrentEvents list of events that overlap. This is a change in how currently this returns a list of all events as a list that match any of the config's events.

Reply:
so it should be more like this.
we care about event X, with causes of Y and Z.

this function should return a list of event Xs where event Xs have a list(relevantEventsType) which are the events happening at the same time that we care about.